server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableNode.java

```diff
@@ -117,9 +118,66 @@ private Locality getLocalityForError() {
    * @param newEndPosition the new end position
    */
   public void extendLocality(Position newEndPosition) {
-    if (RangeUtils.isAfter(locality.getRange().getEnd(), newEndPosition))
+    if (RangeUtils.isBefore(locality.getRange().getEnd(), newEndPosition))
```

Comment:
🤔

Reply:
It was a bug. It's all because of unclear naming.